From 1cd54b5e37730aee3e3c92262241fcdbab4407ac Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Sun, 20 Oct 2019 19:06:19 +0000
Subject: [PATCH] context: Don't fail to create the context if HOME isn't
 available

E.g. when Mutter has CAP_SYS_NICE and thus secure_getenv returns NULL.

Fixes https://bugs.archlinux.org/task/64191
---
 src/context.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/context.c b/src/context.c
index f43ee0a..6d86dd1 100644
--- a/src/context.c
+++ b/src/context.c
@@ -81,13 +81,13 @@ xkb_context_include_path_append_default(struct xkb_context *ctx)
     int ret = 0;
 
     home = secure_getenv("HOME");
-    if (!home)
-        return ret;
-    err = asprintf(&user_path, "%s/.xkb", home);
-    if (err <= 0)
-        return ret;
-    ret |= xkb_context_include_path_append(ctx, user_path);
-    free(user_path);
+    if (home != NULL) {
+        err = asprintf(&user_path, "%s/.xkb", home);
+        if (err <= 0)
+            return ret;
+        ret |= xkb_context_include_path_append(ctx, user_path);
+        free(user_path);
+    }
 
     root = secure_getenv("XKB_CONFIG_ROOT");
     if (root != NULL)
-- 
2.23.0

